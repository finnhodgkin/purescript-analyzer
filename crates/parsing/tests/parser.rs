macro_rules! parser_tests {
    ($($name:ident => $source:expr),+ $(,)?) => {
        $(
            #[test]
            fn $name() {
                let lexed = lexing::lex($source);
                let tokens = lexing::layout(&lexed);
                let (node, errors) = parsing::parse(&lexed, &tokens);
                insta::assert_debug_snapshot!((node, errors));
            }
        )+
    };
}

parser_tests!(
    comment_pre_post => include_str!("parser/CommentPrePost.purs"),
    lexer_error_inserted => include_str!("parser/LexerErrorInserted.purs"),
    module_export => include_str!("parser/ModuleExport.purs"),
    module_export_class => include_str!("parser/ModuleExportClass.purs"),
    module_export_class_error_end => include_str!("parser/ModuleExportClassErrorEnd.purs"),
    module_export_comma_error => include_str!("parser/ModuleExportCommaError.purs"),
    module_export_data => include_str!("parser/ModuleExportData.purs"),
    module_export_data_comma_error => include_str!("parser/ModuleExportDataCommaError.purs"),
    module_export_data_item_after_double_period => include_str!("parser/ModuleExportDataItemAfterDoublePeriod.purs"),
    module_export_data_item_before_double_period => include_str!("parser/ModuleExportDataItemBeforeDoublePeriod.purs"),
    module_export_data_item_invalid => include_str!("parser/ModuleExportDataItemInvalid.purs"),
    module_export_empty => include_str!("parser/ModuleExportEmpty.purs"),
    module_export_invalid => include_str!("parser/ModuleExportInvalid.purs"),
    module_export_type_operator => include_str!("parser/ModuleExportTypeOperator.purs"),
    module_export_type_operator_error => include_str!("parser/ModuleExportTypeOperatorError.purs"),
    module_export_type_operator_error_end => include_str!("parser/ModuleExportTypeOperatorErrorEnd.purs"),
    module_header => include_str!("parser/ModuleHeader.purs"),
    module_header_prefixed => include_str!("parser/ModuleHeaderPrefixed.purs"),
);
