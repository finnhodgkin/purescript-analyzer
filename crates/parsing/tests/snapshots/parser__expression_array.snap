---
source: crates/parsing/tests/parser.rs
expression: "(node, errors)"
snapshot_kind: text
---
(
    Module@0..71
      ModuleHeader@0..30
        MODULE@0..6 "module"
        Comment@6..7
          WHITESPACE@6..7 " "
        ModuleName@7..23
          UPPER@7..22 "ExpressionArray"
          Comment@22..23
            WHITESPACE@22..23 " "
        ModuleExportList@23..23
        WHERE@23..28 "where"
        Comment@28..30
          WHITESPACE@28..30 "\n\n"
      LAYOUT_START@30..30 ""
      ModuleImports@30..30
      ModuleStatements@30..71
        ValueEquation@30..41
          LOWER@30..35 "array"
          Comment@35..36
            WHITESPACE@35..36 " "
          EquationBinders@36..36
          EquationUnconditional@36..41
            EQUAL@36..37 "="
            Comment@37..38
              WHITESPACE@37..38 " "
            EquationWhere@38..41
              ExpressionArray@38..41
                LEFT_SQUARE@38..39 "["
                RIGHT_SQUARE@39..40 "]"
                Comment@40..41
                  WHITESPACE@40..41 "\n"
        LAYOUT_SEPARATOR@41..41 ""
        ValueEquation@41..53
          LOWER@41..46 "array"
          Comment@46..47
            WHITESPACE@46..47 " "
          EquationBinders@47..47
          EquationUnconditional@47..53
            EQUAL@47..48 "="
            Comment@48..49
              WHITESPACE@48..49 " "
            EquationWhere@49..53
              ExpressionArray@49..53
                LEFT_SQUARE@49..50 "["
                ExpressionInteger@50..51
                  INTEGER@50..51 "1"
                RIGHT_SQUARE@51..52 "]"
                Comment@52..53
                  WHITESPACE@52..53 "\n"
        LAYOUT_SEPARATOR@53..53 ""
        ValueEquation@53..71
          LOWER@53..58 "array"
          Comment@58..59
            WHITESPACE@58..59 " "
          EquationBinders@59..59
          EquationUnconditional@59..71
            EQUAL@59..60 "="
            Comment@60..61
              WHITESPACE@60..61 " "
            EquationWhere@61..71
              ExpressionArray@61..71
                LEFT_SQUARE@61..62 "["
                ExpressionInteger@62..63
                  INTEGER@62..63 "1"
                COMMA@63..64 ","
                Comment@64..65
                  WHITESPACE@64..65 " "
                ExpressionInteger@65..66
                  INTEGER@65..66 "2"
                COMMA@66..67 ","
                Comment@67..68
                  WHITESPACE@67..68 " "
                ExpressionInteger@68..69
                  INTEGER@68..69 "3"
                RIGHT_SQUARE@69..70 "]"
                Comment@70..71
                  WHITESPACE@70..71 "\n"
      LAYOUT_END@71..71 ""
      END_OF_FILE@71..71 ""
    ,
    [],
)
